### PR TITLE
Droth 57 area selection by drawn polygon

### PR DESCRIPTION
**Pull request by leirasf**

### CODE REVIEW

File: UI/src/view/SelectToolControl.js [hunk 1]
```diff
@@ -19,13 +22,26 @@
...
+            type: ('Circle'),
+            style: drawStyle(),
+            geometryFunction: ol.interaction.Draw.createBox()
         });
 
         var selectInteraction = new ol.interaction.Select({
             layer: layer,
             condition: function(events){
-                return enabled &&(ol.events.condition.doubleClick(events) || ol.events.condition.singleClick(events));
+                return !isPolygonActive && !isDragBoxActive && enabled &&(ol.events.condition.doubleClick(events) || ol.events.condition.singleClick(events));
```
Comment (leirasf) [inline UI/src/view/SelectToolControl.js]: Maybe we can remove those flag if we set `enabled = false`?

File: UI/src/view/LinearAssetLayer.js [hunk 2]
```diff
@@ -187,7 +187,8 @@ window.LinearAssetLayer = function(params) {
   var selectToolControl = new SelectToolControl(application, vectorLayer, map, {
     style: function(feature){ return feature.setStyle(style.browsingStyleProvider.getStyle(feature, {zoomLevel: uiState.zoomLevel})); },
     onDragEnd: onDragEnd,
-    onSelect: OnSelect
+    onSelect: OnSelect,
```
Comment (leirasf) [inline UI/src/view/LinearAssetLayer.js]: We should change the name to  `onSelect `instead of `OnSelect`.

File: UI/src/view/LinearAssetLayer.js [hunk 3]
```diff
@@ -230,12 +243,24 @@ window.LinearAssetLayer = function(params) {
...
+        selectToolControl.deactivate();
+        linearAssetCutter.activate();
+        break;
+      case 'Select':
+        linearAssetCutter.deactivate();
+        selectToolControl.activate();
+        break;
+      case 'Rectangle':
+        linearAssetCutter.deactivate();
+        selectToolControl.activeDragBox();
```
Comment (MAC-CGI) [inline UI/src/view/LinearAssetLayer.js]: This name "selectToolControl.activeDragBox();" could be changed to "selectToolControl.activeRectangle()"

File: UI/src/view/SelectToolControl.js [hunk 4]
```diff
@@ -4,10 +4,13 @@
         var mapDoubleClickEventKey;
         var enabled = false;
         var initialized = false;
+        var isPolygonActive = false;
+        var isDragBoxActive = false;
```
Comment (MAC-CGI) [inline UI/src/view/SelectToolControl.js]: Could be changed "isDragBoxActive " to "isRectangleActive"

File: UI/src/view/LinearAssetLayer.js [hunk 5]
```diff
@@ -219,6 +221,17 @@ window.LinearAssetLayer = function(params) {
     }
   }
 
+  function onDrawEnd(linearAssets) {
```
Comment (joanacoelho) [inline UI/src/view/LinearAssetLayer.js]: This function should be edited. onDragEnd function on this file has the same code

File: UI/src/view/SelectToolControl.js [hunk 6]
```diff
@@ -43,6 +59,24 @@
             settings.onDragEnd(selectedFeatures);
         });
 
+        function drawEnd(evt) {
+            evt.preventDefault();
```
Comment (joanacoelho) [inline UI/src/view/SelectToolControl.js]: probably the same code in dragBoxInteraction.on('boxend')

**Comment by marquesfm** [comment]
it seems ok

**Pull request merged**